ultravox/tools/ds_tool/ds_tool.py

```diff
@@ -75,10 +97,25 @@ def __post_init__(self):
 
     def map_split(self, ds_split: datasets.Dataset, num_proc: int) -> datasets.Dataset:
         print(f'Generating "{self.new_column_name}" with template:\n{self.template}')
-        return ds_split.map(self._map_sample, num_proc=num_proc)
+        return ds_split.map(
+            self._map_sample, num_proc=num_proc, writer_batch_size=self.write_batch_size
+        )
 
     def _map_sample(self, sample):
-        rendered = jinja2.Template(self.template).render(**sample, json_dump=json.dumps)
+        for field in self.format_fields:
+            sample[field] = format_asr_text(sample[field])
```

Comment:
Could this formatting be done via the jinja template? It does seem kind of specific to this one particular type of completion.

Reply:
I think this is consistent with the selective use of text_proc.format_asr_text in ultravox.data.datasets.py for certain datasets.

Reply:
This formatting is now moved to the jinja template.